### PR TITLE
Api update functionality

### PR DESCRIPTION
**Pull request by JamieZafar**
Users can now update todos by way of a stylish button. This has changed the todo object, which now has an isComplete property. The API is also changed, and uses PUT to replace a todo with an almost identical todo, however the isComplete property has been flipped to true. This changes the font and weight of the todo, so the user can clearly see when it has been marked as complete. Changed the theme of the website to increase synergy. Added unit testing for the new server functionality.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -19,11 +19,24 @@ module.exports = function(port, middleware, callback) {
...
         res.set("Location", "/api/todo/" + todo.id);
         res.sendStatus(201);
     });
 
+    // Update
+    app.put("/api/todo/:id", function(req, res) {
+        var id = req.params.id;
+        var todo = getTodo(id);
+        if (todo) {
+            todo.isComplete = true;
```
Comment (cuthullu) [inline server/server.js]: Can this be set by property in the request body so we can un-mark them complete too

File: server/server.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ module.exports = function(port, middleware, callback) {
         var id = req.params.id;
         var todo = getTodo(id);
         if (todo) {
-            todo.isComplete = true;
+            todo.isComplete = !todo.isComplete;
```
Comment (cuthullu) [inline server/server.js]: What happens if you have two clients connected and updating the same todo?

File: public/main.js [hunk 3]
```diff
@@ -52,13 +52,24 @@ function reloadTodoList() {
...
+            var completeBtn = document.createElement("button");
             listItem.textContent = todo.title;
             deleteBtn.textContent = "Delete";
-            deleteBtn.style.background = "black";
-            deleteBtn.style.color = "red";
+            completeBtn.textContent = "Complete";
+            deleteBtn.className = "deleteButton";
+            completeBtn.className = "completeButton";
+            if (todo.isComplete) {
+                listItem.style.color = "red";
```
Comment (kenneth-gray) [inline public/main.js]: No inline styles please - use a css class

File: public/main.js [hunk 4]
```diff
@@ -52,13 +52,24 @@ function reloadTodoList() {
         todos.forEach(function(todo) {
             var listItem = document.createElement("li");
             var deleteBtn = document.createElement("button");
+            var completeBtn = document.createElement("button");
```
Comment (kenneth-gray) [inline public/main.js]: The complete button and delete button setup is spread out? Could you group like for like and then factor out a function for creating buttons please?

File: public/main.js [hunk 5]
```diff
@@ -52,13 +52,22 @@ function reloadTodoList() {
...
             listItem.textContent = todo.title;
             deleteBtn.textContent = "Delete";
-            deleteBtn.style.background = "black";
-            deleteBtn.style.color = "red";
+            completeBtn.textContent = "Complete";
+            deleteBtn.className = "deleteButton";
+            completeBtn.className = "completeButton";
+            if (todo.isComplete) {
+                listItem.className = "listItem";
```
Comment (kenneth-gray) [inline public/main.js]: listItem? How about complete or completed?

File: public/main.js [hunk 6]
```diff
@@ -51,14 +57,21 @@ function reloadTodoList() {
...
         todos.forEach(function(todo) {
             var listItem = document.createElement("li");
-            var deleteBtn = document.createElement("button");
+            var deleteBtn = makeButton("Delete");
+            var completeBtn = makeButton("Complete");
             listItem.textContent = todo.title;
-            deleteBtn.textContent = "Delete";
-            deleteBtn.style.background = "black";
-            deleteBtn.style.color = "red";
+            deleteBtn.className = "deleteButton";
```
Comment (kenneth-gray) [inline public/main.js]: Could these 2 not go in the makeButton? How about the event listeners too?

File: public/main.js [hunk 7]
```diff
@@ -51,14 +58,19 @@ function reloadTodoList() {
...
             listItem.textContent = todo.title;
-            deleteBtn.textContent = "Delete";
-            deleteBtn.style.background = "black";
-            deleteBtn.style.color = "red";
+            if (todo.isComplete) {
+                listItem.className = "completed-item";
+            }
             deleteBtn.addEventListener("click", function() {
                 deleteTodo(todo);
             });
```
Comment (lclarksl) [inline public/main.js]: Can you set the event listener in the `makeButton` function too?

File: server/server.js [hunk 8]
```diff
@@ -19,11 +19,25 @@ module.exports = function(port, middleware, callback) {
...
+        todo.isComplete = false;
         todos.push(todo);
         res.set("Location", "/api/todo/" + todo.id);
         res.sendStatus(201);
     });
 
+    // Update
+    app.put("/api/todo/:id", function(req, res) {
+        var id = req.params.id;
+        var complete = req.params.complete;
```
Comment (kenneth-gray) [inline server/server.js]: This isn't used any more

**Comment by kenneth-gray** [comment]
:shipit:

**Pull request merged**